recipes/glib/all/conanfile.py

```diff
@@ -90,91 +93,108 @@ def requirements(self):
             # for Linux, gettext is provided by libc
             self.requires("libgettext/0.21")
 
-        if tools.is_apple_os(self.settings.os):
+        if is_apple_os(self.settings.os):
             self.requires("libiconv/1.17")
 
     def validate(self):
         if hasattr(self, 'settings_build') and cross_building(self, skip_x64_x86=True):
```

Comment:
@SSE4 I was trying out this change on a macOS m1 using docker (so we cross-compile inside a docker container), and it seems like this check prevents cross-compiling. If the condition is removed, it compiles just fine.

We get:
```
#11 4.084 Installing (downloading, building) binaries...
#11 4.084 ERROR: There are invalid packages (packages that cannot exist for this configuration):
#11 4.084 glib/2.73.3@tctl/dev: Invalid ID: Cross-building not implemented
#11 4.114 CMake Error at /build/tctl-cmake/externals/conan.cmake:631 (message):
```

Could it be changed to allow cross-compiling (we removed the condition for now)?

Reply:
yes, I think it should be allowed now